engine/src/cubos/engine/voxels/palette.cpp

```diff
@@ -100,6 +100,18 @@ uint16_t VoxelPalette::add(const VoxelMaterial& material, float similarity)
     return this->size();
 }
 
+uint16_t VoxelPalette::push(const VoxelMaterial& material)
+{
+    if (this->size() == UINT16_MAX)
+    {
+        CUBOS_ERROR("Cannot add new material: palette is full");
+        return -1;
```

Comment:
```suggestion
        CUBOS_FAIL("Cannot add new material: palette is full");
```

Aborting here is okay, but we should add a `@note` to the docs warning of it.